gordo_components/client/client.py

```diff
@@ -390,28 +398,38 @@ def predict(
         PredictionResult
         """
 
-        json = {
-            "X": server_utils.dataframe_to_dict(X.iloc[chunk]),
-            "y": server_utils.dataframe_to_dict(y.iloc[chunk])
-            if y is not None
-            else None,
-        }
+        kwargs: Dict[str, Any] = dict(
+            session=session,
+            url=f"{endpoint.endpoint}{self.prediction_path}{self.query_params}",
+        )
 
+        # We're going to serialize the data as either JSON or MessagePack
+        if self.use_pyarrow:
+            kwargs["files"] = {
+                "X": pa.serialize_pandas(X.iloc[chunk]).to_pybytes(),
```

Comment:
It is my impression that this does not do any compression, but is pretty much a pure memory dump. Which will still provide smaller messages since it is pure bytes, instead of text. But we can compress by using the pyarrow parquet writer and use its compression (default is "snappy"). An example is [here](https://arrow.apache.org/docs/python/parquet.html#reading-and-writing-single-files). The "file" can be a "[NativeFile](https://arrow.apache.org/docs/python/memory.html#input-and-output)", of which "BufferOutputStream" is one of the implementations, so sounds like getting the bytestream should be pretty easy.